provisioning/service/src/provisioningserviceclient.ts

```diff
@@ -277,11 +317,16 @@ export class ProvisioningServiceClient {
     });
   }
 
-  private _delete(endpointPrefix: string, enrollmentOrIdOrRegistration: string | any, etagOrCallback?: string | DeleteCallback, deleteCallback?: DeleteCallback): void {
+  private _delete(endpointPrefix: string, enrollmentOrIdOrRegistration: string | any, etagOrCallback?: string | ErrorCallback, deleteCallback?: ErrorCallback): void {
     let ifMatch: string;
-    let suppliedCallback: DeleteCallback;
+    let suppliedCallback: ErrorCallback;
     let id: string;
 
+    suppliedCallback = deleteCallback || (etagOrCallback instanceof Function ? etagOrCallback : undefined);
+    if (!suppliedCallback) {
+      throw new ArgumentError('No callback was passed.');
+    }
```

Comment:
This would appear to be a breaking change.  I believe the original code allowed the caller NOT have the callback defined.

Reply:
@anthonyvercolano

Reply:
If the user doesn't specify a callback and an error occurs than I think the application developer has indicated that they just don'e care.  Bad idea, but not the SDK's problem.

Reply:
I see. How do you think it should work here? Add a flag we want a Promise to be returned?